rfq/stream.go

```diff
@@ -92,8 +92,15 @@ func NewStreamHandler(ctx context.Context,
 func (h *StreamHandler) handleIncomingWireMessage(
 	wireMsg rfqmsg.WireMessage) error {
 
+	// Define a function to look up an RFQ session. It is necessary to
+	// utilize the corresponding outgoing request when parsing some
+	// incoming wire messages.
+	sessionLookup := func(id rfqmsg.ID) (rfqmsg.OutgoingMsg, bool) {
+		return h.outgoingRequests.Load(id)
+	}
+
 	// Parse the wire message as an RFQ message.
-	msg, err := rfqmsg.NewIncomingMsgFromWire(wireMsg)
+	msg, err := rfqmsg.NewIncomingMsgFromWire(wireMsg, sessionLookup)
```

Comment:
Could we directly pass in a reference to `h.outgoingRequests.Load` here, instead of creating the `sessionLookup` wrapper? Or does that not work because of type assertions?

Reply:
that works! thanks for the suggestion